src/commands/lambda/functions/instrument.ts

```diff
@@ -184,7 +184,7 @@ export const calculateUpdateRequest = async (
   }
 
   // Update Python Handler
-  if (runtimeType === RuntimeType.PYTHON && settings.layerVersion !== undefined) {
+  if (runtimeType === RuntimeType.PYTHON && (settings.layerVersion !== undefined || settings.interactive)) {
```

Comment:
Why were we checking that `settings.layerVersion !== undefined` before?

Reply:
Because sometimes, user might only want to instrument their AWS Lambda w/o changing their current layer version, or they simply don't want to add it.

When using the Layer, we need to change the handler, because that's how we are able to provide monitoring.